staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -57,11 +57,13 @@ func IsListType(obj runtime.Object) bool {
 		ok = err == nil
 
 		// cache only the first 1024 types
-		isListCache.lock.Lock()
 		if len(isListCache.byType) < 1024 {
```

Comment:
len() is not thread safe for map. reference: https://stackoverflow.com/questions/52710882/is-len-thread-safe-in-golang

Reply:
There is a second check on line 62 below.
If the value of length reaches 1024 here, it would never dip below 1024 regardless of when the next check is.

Reply:
You cannot assume anything on reading an integer while somebody else is updating it. Just an example:

let say the value is 0xFF 0x03 (little endian system). If a user wants to update it to 0x00 0x04 and it only managed to write the first byte and got context switched (the memory is 0x00 0x03) the other process may proceed as this number is less than 1024 but the actual length is 1024. This is just an example and I didn't mention it so we can fix it as there could be many others. We can't simply assume len is safe outside of a lock in any way.